_posts/2021-02-18-survey/survey.Rmd

```diff
@@ -0,0 +1,314 @@
+
+[^6]: This question only applied Spark users working in R.
+
+```{r, eval = TRUE, echo = FALSE, layout="l-body-outset", fig.cap="Self-rated skills re R and Spark."}
+knitr::include_graphics("images/spark_skills.png")
+```
+
+### Wishes and suggestions
+
+Just like with DL, Spark users were asked what could be improved, and what they were hoping for.
+
+Interestingly, answers are less "clustered" than for DL. While with DL, a few things cropped up again and again, and there were very few mentions of concrete technical features, here we see about the opposite: The great majority of wishes are concrete, technical, and often, have been uttered just a single time.
+
+Probably though, this is not a coincidence.
+
+Looking back at how `sparklyr` has evolved from 2016 until now, there is a persistent theme of it being the bridge that joins the (vast in itself!) Apache Spark ecosystem to numerous useful R interfaces, frameworks, and utilities (most notably, the tidyverse).
+
+Many of our users' suggestions are essentially a continuation of this theme. This holds, for example, for two features already available as of `sparklyr` 1.4 and 1.2, respectively: support for the Arrow serialization format and for Databricks Connect. It also holds for `tidymodels` integration (a frequent wish), a simple R interface for defining Spark UDFs (frequently desired, this one too), out-of-core direct computations on Parquet files, and extended time series functionalities.
+
+We're thankful for the feedback and will evaluate carefully what could be done in each case. In general, integrating `sparklyr` with some feature X is a process to be planned carefully, as modifications could, in theory, be made in various places (`sparklyr`; X; both `sparklyr` and X; or even a newly-to-be-created extension). In fact, this is a topic deserving of much more detailed coverage, and has to be left to a future post.
+
+Now to the last section, that on ethics and societal impact of AI.
+
+# Ethics and AI in society
+
+To start, this is probably the section that will profit most from more preparation, the next time we do this survey. Due to time pressure, some (not all!) of the questions ended up being too suggestive, possibly resulting in non-insignificant social desirability bias.[^7]
+
+[^7]: I'll indicate below the questions for which I think this might have been the case.
+
+Next time, thus, I'll try to avoid this, and probably questions in this area will look pretty different (more like scenarios or what-if stories)[^8]. However, I was told by several people they'd been positively surprised by simply *encountering this topic at all in the survey*. So perhaps this is the main point -- although there are a few results that I'm sure will be interesting by themselves!
```

Comment:
```suggestion
Next time, we'll try to avoid this, and questions in this area will likely look pretty different (more like scenarios or what-if stories)[^8]. However, I was told by several people they'd been positively surprised by simply *encountering this topic at all in the survey*. So perhaps this is the main point -- although there are a few results that I'm sure will be interesting by themselves!
```